src/Symfony/Component/WebLink/HttpHeaderParser.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\WebLink;
+
+/**
+ * Parse a list of HTTP Link headers into a list of Link instances.
+ *
+ * @see https://tools.ietf.org/html/rfc5988
+ *
+ * @author Jérôme Tamarelle <jerome@tamarelle.net>
+ */
+final class HttpHeaderParser
```

Comment:
final + no base type = not SOLID

Reply:
I blindly replicated the same thing as `HttpHeaderSerializer`. Let's remove the `final`.

Reply:
We should remove final there also then.
BTW, did you consider a service definition to allow autowiring?

Reply:
I removed the `final` from the other class and added framework integration.

Reply:
Should we add an interface to provider autowiring based on the interface rather than the concrete implementation ?

Reply:
In absolute terms, that's what we should do. We would do the same for `HttpHeaderSerializer` then.
But I don't see why anyone would write a different implementation.